testnet/stacks-node/src/tests/signer/v0.rs

```diff
@@ -223,45 +259,23 @@ fn block_proposal_rejection() {
     let short_timeout = Duration::from_secs(30);
 
     info!("------------------------- Send Block Proposal To Signers -------------------------");
-    let miners_contract_id = boot_code_id(MINERS_NAME, false);
-    let mut session = StackerDBSession::new(
-        &signer_test.running_nodes.conf.node.rpc_bind,
-        miners_contract_id.clone(),
-    );
-    let block = NakamotoBlock {
+    let reward_cycle = signer_test.get_current_reward_cycle();
+    let view = SortitionsView::fetch_view(&signer_test.stacks_client).unwrap();
+    let mut block = NakamotoBlock {
         header: NakamotoBlockHeader::empty(),
         txs: vec![],
     };
+    // We want to force the signer to submit the block to the node for validation
+    // Must set the pox treatment validly and consensus hash validlty to prevent early termination
+    block.header.pox_treatment = BitVec::ones(1).unwrap();
+    block.header.consensus_hash = view.cur_sortition.consensus_hash;
+
```

Comment:
I think this modification should be a second block proposal. That way, the test covers both a rejection by the chainstate module and by the proposal endpoint.

Reply:
The test above covers this other test case but I will combine the two so its more concise/less duplication!